#!/usr/bin/env python
PACKAGE = "hardware_utils"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("ATC_ANG_RLL_P", double_t, 0, "ATC_ANG_RLL_P", 0.0, 0.0, 10.0)
gen.add("ATC_ANG_PIT_P", double_t, 0, "ATC_ANG_PIT_P", 0.0, 0.0, 10.0)
gen.add("ATC_ANG_YAW_P", double_t, 0, "ATC_ANG_YAW_P", 0.0, 0.0, 10.0)
gen.add("ATC_RAT_RLL_P", double_t, 0, "ATC_RAT_RLL_P", 0.0, 0.0, 0.30)
gen.add("ATC_RAT_RLL_I", double_t, 0, "ATC_RAT_RLL_I", 0.0, 0.0, 0.30)
gen.add("ATC_RAT_RLL_D", double_t, 0, "ATC_RAT_RLL_D", 0.0, 0.0, 0.01)
gen.add("ATC_RAT_PIT_P", double_t, 0, "ATC_RAT_PIT_P", 0.0, 0.0, 0.30)
gen.add("ATC_RAT_PIT_I", double_t, 0, "ATC_RAT_PIT_I", 0.0, 0.0, 0.30)
gen.add("ATC_RAT_PIT_D", double_t, 0, "ATC_RAT_PIT_D", 0.0, 0.0, 0.01)
gen.add("ATC_RAT_YAW_P", double_t, 0, "ATC_RAT_YAW_P", 0.0, 0.0, 0.30)
gen.add("ATC_RAT_YAW_I", double_t, 0, "ATC_RAT_YAW_I", 0.0, 0.0, 0.30)
gen.add("ATC_RAT_YAW_D", double_t, 0, "ATC_RAT_YAW_D", 0.0, 0.0, 0.01)
gen.add("PSC_POSXY_P", double_t, 0, "PSC_POSXY_P", 0.0, 0.0, 2.0)
gen.add("PSC_POSZ_P", double_t, 0, "PSC_POSZ_P", 0.0, 0.0, 4.0)
gen.add("PSC_VELXY_P", double_t, 0, "PSC_VELXY_P", 0.0, 0.0, 5.0)
gen.add("PSC_VELXY_I", double_t, 0, "PSC_VELXY_I", 0.0, 0.0, 5.0)
gen.add("PSC_VELZ_P", double_t, 0, "PSC_VELZ_P", 0.0, 0.0, 10.0)
gen.add("PSC_VELZ_I", double_t, 0, "PSC_VELZ_I", 0.0, 0.0, 5.0)
gen.add("PSC_ACCZ_P", double_t, 0, "PSC_ACCZ_P", 0.0, 0.0, 1.0)
gen.add("PSC_ACCZ_I", double_t, 0, "PSC_ACCZ_I", 0.0, 0.0, 1.0)
gen.add("PSC_ACCZ_D", double_t, 0, "PSC_ACCZ_D", 0.0, 0.0, 1.0)
# gen.add("VISO_DELAY_MS", double_t, 0, "VISO_DELAY_MS", 0.0, 0.0, 250.0)

exit(gen.generate(PACKAGE, "hard_mavros_bridge", "PID"))